kettle-plugins/common/ui/src/main/java/org/pentaho/big/data/plugins/common/ui/MultishimNamedCluster.java

```diff
@@ -0,0 +1,188 @@
+    }
+
+
+    return nc.getName();
+  }
+
+  private void saveNamedCluster( IMetaStore metaStore, NamedCluster namedCluster ) {
+    try {
+      namedClusterService.create( namedCluster, metaStore );
+    } catch ( MetaStoreException e ) {
+      /*commonDialogFactory.createErrorDialog( spoon.getShell(),
+          BaseMessages.getString( PKG, SPOON_DIALOG_ERROR_SAVING_NAMED_CLUSTER_TITLE ),
+          BaseMessages.getString( PKG, SPOON_DIALOG_ERROR_SAVING_NAMED_CLUSTER_MESSAGE, namedCluster.getName() ), e );*/
+    }
+  }
+
+
+  private void addConfigProperties( NamedCluster namedCluster ) throws Exception {
+    Path clusterConfigDirPath = Paths.get( getNamedClusterConfigsRootDir( null ) + "/" + namedCluster.getConfigId() );
+    Path configPropertiesPath = Paths.get( getNamedClusterConfigsRootDir( null ) + "/" + namedCluster.getConfigId() + "/" + "config.properties" );
+    Files.createDirectories( clusterConfigDirPath );
+    String sampleConfigProperties = namedCluster.getShimIdentifier() + "sampleconfig.properties";
+    InputStream inputStream = this.getClass().getClassLoader().getResourceAsStream( sampleConfigProperties );
+    if ( inputStream != null ) {
+      Files.copy( inputStream, configPropertiesPath, StandardCopyOption.REPLACE_EXISTING );
+    }
+  }
+
+
+  public void delNamedCluster( IMetaStore metaStore, NamedCluster namedCluster ) {
+    if ( metaStore == null ) {
+      metaStore = spoon.getMetaStore();
+    }
+    deleteNamedCluster( metaStore, namedCluster );
+    spoon.refreshTree( STRING_NAMED_CLUSTERS );
+    spoon.setShellText();
+  }
+
+  private XmlMetaStore getXmlMetastore( IMetaStore metaStore ) throws MetaStoreException {
+    XmlMetaStore xmlMetaStore = null;
+
+    if ( metaStore instanceof DelegatingMetaStore ) {
+      IMetaStore activeMetastore = ( (DelegatingMetaStore) metaStore ).getActiveMetaStore();
+      if ( activeMetastore instanceof XmlMetaStore ) {
+        xmlMetaStore = (XmlMetaStore) activeMetastore;
+      }
+    } else if ( metaStore instanceof XmlMetaStore ) {
+      xmlMetaStore = (XmlMetaStore) metaStore;
+    }
+
+    return xmlMetaStore;
+  }
+
+
+  private void deleteNamedCluster( IMetaStore metaStore, NamedCluster namedCluster ) {
+    try {
+      if ( namedClusterService.read( namedCluster.getName(), metaStore ) != null ) {
+        namedClusterService.delete( namedCluster.getName(), metaStore );
+        XmlMetaStore xmlMetaStore = getXmlMetastore( metaStore );
+        if ( xmlMetaStore != null ) {
+          String path = getNamedClusterConfigsRootDir( xmlMetaStore ) + "/" +  namedCluster.getConfigId();
+          try {
+            FileUtils.deleteDirectory( new File( path ) );
+          } catch ( IOException e ) {
+            // Do nothing. The config directory will be orphaned but functionality will not be impacted.
+          }
+        }
+      }
+    } catch ( MetaStoreException e ) {
+      /*commonDialogFactory.createErrorDialog( spoon.getShell(),
```

Comment:
Can these comments be deleted, or are they there for documentation purposes? Should probably do something with this exception, write it to console output if not a log file.

Reply:
Peter. I left those comments as a reference for now. That's what the current code based on the SWT UI does. I will know more about what to do with those exceptions as James progresses on the UI front.